contracts/AnchoredPriceView.sol

```diff
@@ -0,0 +1,302 @@
+    }
+
+    /**
+     * @notice Primary entry point to post and recalculate prices
+     * @dev We let anyone pay to post anything, but only prices from configured source will be stored in the view
+     * @param messages The messages to post to the oracle
+     * @param signatures The signatures for the corresponding messages
+     */
+    function postPrices(bytes[] calldata messages, bytes[] calldata signatures, string[] calldata symbols) external {
+        require(messages.length == signatures.length, "messages and signatures must be 1:1");
+
+        // Save the prices
+        for (uint i = 0; i < messages.length; i++) {
+            priceData.put(messages[i], signatures[i]);
+        }
+
+        // load usdc for using in loop to convert anchor prices to dollars
+        uint256 usdcPrice = anchor.getUnderlyingPrice(cUsdcAddress);
+
+        // Try to update the view storage
+        for (uint i = 0; i < symbols.length; i++) {
+            string memory symbol = symbols[i];
+            address tokenAddress = getCTokenAddress(symbol);
+            uint64 sourcePrice = priceData.getPrice(source, symbol);
+            uint64 anchorPrice = getAnchorPrice(tokenAddress, usdcPrice);
+
+            if (anchorPrice == 0 || tokenAddress == cUsdcAddress || tokenAddress == cUsdtAddress) {
+                emit PriceGuarded(symbol, sourcePrice, anchorPrice);
+            } else {
+                uint256 anchorRatioMantissa = uint256(sourcePrice) * 100e16 / anchorPrice;
+                // Only update the view's price if the source price is within a bound, and it is a new median
+                if (anchorRatioMantissa <= upperBoundAnchorRatio && anchorRatioMantissa >= lowerBoundAnchorRatio) {
+                    // only update and emit event if the source price is new, otherwise do nothing
+                    if (prices[symbol] != sourcePrice) {
+                        prices[symbol] = sourcePrice;
+                        emit PriceUpdated(symbol, sourcePrice);
+                    }
+                } else {
+                    emit PriceGuarded(symbol, sourcePrice, anchorPrice);
+                }
+            }
+        }
+    }
+
+    /**
+     * @notice Flags that this contract is meant to be compatible with Compound v2 PriceOracle interface.
+     * @return true, this contract is meant to be used as Compound v2 PriceOracle interface.
+     */
+    function isPriceOracle() external pure returns (bool) {
+        return true;
+    }
+
+    /**
+     * @dev fetch price in eth from proxy and convert to usd price using anchor usdc price.
+     * @dev Anchor usdc price has 30 decimals, and anchor general price has 18 decimals, so multiplying by 1e18 and by 1e30 yields 1e6
+     */
+    function getAnchorPrice(address tokenAddress, uint256 usdcPrice) public view returns (uint64) {
+
+        if ( tokenAddress == cUsdcAddress || tokenAddress == cUsdtAddress )  {
+            // hard code to 1 dollar
+            return uint64(oneDollar);
```

Comment:
Oh actually this is pretty confusing - we pass in the `usdcPrice` but its not the `usdcPrice`?

Reply:
yes, usdcPrice ( in terms of Ether ) is used to convert other ether prices to dollars, but if actually wanting anchor price for usdc, we should always return 1 dollar. 

There is a little oddity here that we probably shouldn't even really be trying to post a price for usdc or usdt, but I thought we would want to guard against it in some form

Reply:
Still think we should rename input `usdcPrice` to `usdcPriceInEther` or something